Update deprecated asserThat with MatcherAssert

Chinex-Boroja opened the pull request:
## This is a PR for issue [445](https://github.com/odk-x/tool-suite-X/issues/445), and the PR contains the following:

1. Updated the test dependencies in the `androidlibrary_lib/build.gradle`
2. Replaced the deprecated `InstrumentationRegistry.getContext()` with `androidx.test.platform.app.InstrumentationRegistry.getInstrumentation().getTargetContext();` in `PropertiesNonPrivilegedTest` class
3. Updated the deprecated `assertThat()` with `MatcherAssert()` method call in `AlertNProgressMsgFragmentMgerTest` class
4. Handled possible NullPointerExceptions that might be thrown using `Objects.requireNonNull()` in both `testAlertDialogIsCreated()` and `testProgressDialogIsCreated()` methods in the `AlertNProgressMsgFragmentMgerTest` class

### Screenshots of the test results

1.  Unit test of `testAlertDialogIsCreated` method
![UnitTest of testAlertDialogIsCreated](https://github.com/odk-x/androidlibrary/assets/61055200/e7939cbf-f786-4f97-9ee2-347f93d92ef2)

2. Unit test of `testProgressDialogIsCreated` method
![UnitTest of testProgressDialogIsCreated](https://github.com/odk-x/androidlibrary/assets/61055200/8d947544-c1c1-474c-a884-f6f6d040b604)

3. Integrated Test of the `AlertNProgressMsgFragmentMgerTest`
![IntegratedTest Of AlertProgressMsgFragmentMgerTest](https://github.com/odk-x/androidlibrary/assets/61055200/c332da13-5c9c-446d-827d-1ba36789eecd)

4. AndroidTest of AndroidLibrary
![IntegratedTest Of AndroidLibrary](https://github.com/odk-x/androidlibrary/assets/61055200/32e7ab10-0f6b-4141-98cd-1c73a944ed65)

wbrunette commented on androidlibrary_lib/src/androidTest/java/org/opendatakit/logic/PropertiesNonPrivilegedTest.java:
```diff
@@ -64,7 +63,7 @@ public void setUp() throws Exception {
     public void testSimpleProperties() {
 
-        Context context = InstrumentationRegistry.getContext();
+        Context context = androidx.test.platform.app.InstrumentationRegistry.getInstrumentation().getTargetContext();
```
You should add a new import for "androidx.test.platform.app" that replaces the old import instead of writing it out several times. Imports are used to not have to type out the full class package every time.

Chinex-Boroja replied:
> You should add a new import for "androidx.test.platform.app" that replaces the old import instead of writing it out several times. Imports are used to not have to type out the full class package every time.

Hi @wbrunette I have pushed the changes as requested for the new imports

wbrunette reviewed (changes requested):
please remove the "colors.xml" and "theme.xml". These likely are from the UI update branches as they are part of the revised UI. We are not ready to move the changes in the revised UI into the other branches yet.

Chinex-Boroja commented:
> please remove the "colors.xml" and "theme.xml". These likely are from the UI update branches as they are part of the revised UI. We are not ready to move the changes in the revised UI into the other branches yet.

I have done that @wbrunette 
Thank you

wbrunette commented:
@Chinex-Boroja sorry for not being more clear. What I meant was remove the changes to the files "colors.xml" and "theme.xml" as the files still need to exist in their unchanged stated. Your current PR will remove them and they are needed for the code to work.

Chinex-Boroja commented:
> @Chinex-Boroja sorry for not being more clear. What I meant was remove the changes to the files "colors.xml" and "theme.xml" as the files still need to exist in their unchanged stated. Your current PR will remove them and they are needed for the code to work.

I have reverted the files and removed the changes. Kindly review again @wbrunette. Thank you

wbrunette commented on androidlibrary_lib/src/androidTest/java/org/opendatakit/fragment/AlertNProgessMsgFragmentMgerTest.java:
```diff
@@ -61,11 +62,10 @@ public void testAlertDialogIsCreated() {
-                Fragment frag = fragmentManager.findFragmentByTag(ALERT_DIALOG_TAG);
-                assertNotNull(frag);
-                assertThat(frag.isAdded(), is(true));
+                MatcherAssert.assertThat(Objects.requireNonNull(fragmentManager.findFragmentByTag(ALERT_DIALOG_TAG)).isAdded(), is(true));
```
This is a more minor point, but changing from assertNotNull to using Objects.requireNonNull is changing the output of the test. When asserting, the test is applying a check... combining the check into a longer statement makes it less obvious where the problem is. AssertNotNull will cause a failure and report the exact problem.

I do not think it's an improvement, so I kindly request you restore assertNotNull

Chinex-Boroja replied:
> This is a more minor point, but changing from assertNotNull to using Objects.requireNonNull is changing the output of the test. When asserting, the test is applying a check... combining the check into a longer statement makes it less obvious where the problem is. AssertNotNull will cause a failure and report the exact problem.
> 
> I do not think it's an improvement, so I kindly request you restore assertNotNull

Hi @wbrunette I have changed it back. Thank you for the insight

wbrunette commented on androidlibrary_lib/src/androidTest/java/org/opendatakit/fragment/AlertNProgessMsgFragmentMgerTest.java:
```diff
@@ -78,11 +78,11 @@ public void testProgressDialogIsCreated() {
 
                 systemUnderTest.createProgressDialog(PROGRESS_DIALOG_TITLE, PROGRESS_DIALOG_MESSAGE, fragmentManager);
                 fragmentManager.executePendingTransactions();
-                Fragment frag = fragmentManager.findFragmentByTag(PROGRESS_DIALOG_TAG);
```
same here. testing in small steps helps pinpoint issues quickly.

Chinex-Boroja replied:
> same here. testing in small steps helps pinpoint issues quickly.

done too.

wbrunette reviewed (approved):
This looks good now. I am not going to merge it until after the contribution period as the updates to the versions will propagate to all PRs that other outreachy applicants have open.

wbrunette commented:
You can just put a link to the PR as your contribution.

Chinex-Boroja commented:
> You can just put a link to the PR as your contribution.

Thank you
